cmds/tui.c

```diff
@@ -291,7 +293,37 @@ static void print_graph_self(struct field_data *fd)
 	struct uftrace_graph_node *node = fd->arg;
 	uint64_t d;
 
-	d = node->total_time.sum - node->child_time;
+	d = node->self_time.sum;
+
+	print_time(d);
+}
+
+static void print_graph_self_avg(struct field_data *fd)
+{
+	struct uftrace_graph_node *node = fd->arg;
+	uint64_t d;
+
+	d = node->self_time.sum / node->nr_calls;
```

Comment:
You need to check if `node->nr_calls` is 0.

Reply:
No, nr_calls can be any number here.  You should not add ASSERT here.  Just check if it's zero to prevent a divide-by-zero.

Reply:
Oh, that'sure